nvflare/apis/fl_context.py

```diff
@@ -120,59 +192,64 @@ def remove_prop(self, key: str):
             self.props.pop(key, None)
 
     def __str__(self):
-        raw_list = [f'{k}: {type(v["value"])}' for k, v in self.props.items()]
+        raw_list = [f"{k}: {type(v[V])}" for k, v in self.props.items()]
         return " ".join(raw_list)
 
     # some convenience methods
-    def get_engine(self):
-        return self.get_prop(key=ReservedKey.ENGINE, default=None)
+    def _simple_get(self, key: str, default=None):
```

Comment:
The original one has a lock, but _simple_get does not have lock.  Is it OK to skip the lock?

Reply:
The only thing that really need lock in dict is looping thru the dict (in case another thread is adding/deleting from the dict). Simple get is thread safe.